tests/test_bpo.py

```diff
@@ -291,3 +291,66 @@ def __init__(self, *, getitem=None):
     await bpo.router.dispatch(event, gh)
     assert gh.patch_data is None
     assert gh.patch_url is None
+
+
+@pytest.mark.asyncio
+async def test_set_comment_body_success():
+    data = {
+        "action": "opened",
```

Comment:
You're only testing for the pull_request opened event.
Can you also test for the pull_request edited, issue_comment created, and issue_comment edited events?
Probably you can use pytest.mark.parametrize for it.

And please do the same for the other two tests below.

Reply:
Thanks! pytest.mark.parametrize was great!
I've used the `changes` key in the `pull_request` event's data for both action `opened` and `edited`. It keeps the both the cases to be used from single function. I hope it wont be a problem :slightly_smiling_face: